app/components/combined_system_message/combined_system_message.js

```diff
@@ -304,38 +304,51 @@ export default class CombinedSystemMessage extends React.PureComponent {
         );
     }
 
+    renderMessage(postType, userIds, actorId, style) {
+        return (
+            <React.Fragment key={postType + actorId}>
+                {this.renderFormattedMessage(postType, userIds, actorId, {baseText: style.baseText, linkText: style.linkText})}
+            </React.Fragment>
+        );
+    }
+
     render() {
         const {
+            currentUserId,
             messageData,
             theme,
         } = this.props;
         const style = getStyleSheet(theme);
 
         const content = [];
+        const removedUserIds = [];
         for (const message of messageData) {
             const {
                 postType,
                 actorId,
             } = message;
             let userIds = message.userIds;
 
-            if (!this.props.showJoinLeave && actorId !== this.props.currentUserId) {
-                const affectsCurrentUser = userIds.indexOf(this.props.currentUserId) !== -1;
+            if (!this.props.showJoinLeave && actorId !== currentUserId) {
+                const affectsCurrentUser = userIds.indexOf(currentUserId) !== -1;
 
                 if (affectsCurrentUser) {
                     // Only show the message that the current user was added, etc
-                    userIds = [this.props.currentUserId];
+                    userIds = [currentUserId];
                 } else {
                     // Not something the current user did or was affected by
                     continue;
                 }
+            } else if (postType === REMOVE_FROM_CHANNEL) {
```

Comment:
I think this if statement shouldn't be an else-if. If another user removes the current user from the channel and the current user later rejoins, then the "current user was removed from channel" message won't be grouped properly because the code in the first if statement will run, and this branch will be skipped.

Reply:
You're right.  Updated similar to webapp.  Thanks!